### PR TITLE
pscanrules: add website links to help pages 

### PR DESCRIPTION
**Pull request by PariharAditya**
Signed-off-by: Aditya Parihar <97739299+PariharAditya@users.noreply.github.com>

## Overview
Implemented a new interface for linking to website
updated help with there respective ID

## Related Issues
zaproxy/zaproxy#8189

## Checklist
- [x] Update help
- [x] Update changelog
- [x] Run `./gradlew spotlessApply` for code formatting
- [ ] Write tests
- [ ] Check code coverage
- [x] Sign-off commits
- [x] Squash commits
- [x] Use a descriptive title

For more details, please refer to the [developer rules and guidelines](https://www.zaproxy.org/docs/developer/dev-rules-and-guidelines/).

### CODE REVIEW

**Comment by kingthorin** [comment]
This is missing the help half of the change 😉

File: addOns/pscanrules/src/main/javahelp/org/zaproxy/zap/extension/pscanrules/resources/help/contents/pscanrules.html [hunk 1]
```diff
@@ -409,34 +488,45 @@ <H2>VIEWSTATE</H2>
...
 This scan rule check for the Anti-MIME-Sniffing header X-Content-Type-Options and ensures it is set to 'nosniff'.<br>
 At MEDIUM and HIGH thresholds this rule does not alert on client or server error responses or redirects.<br> 
 At LOW threshold it will alert on all responses including errors and redirects.<br>
 <p>
 Latest code: <a href="https://github.com/zaproxy/zap-extensions/blob/main/addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/XContentTypeOptionsScanRule.java">XContentTypeOptionsScanRule.java</a>
+<br>
+Alert ID: <a href="https://www.zaproxy.org/docs/alerts/10021/">10021</a>.
 
-<H2>X-Debug-Token Information Leak</H2>
+<H2 id = "id-10052">X-Debug-Token Information Leak</H2>
```
Comment (kingthorin) [inline addOns/pscanrules/src/main/javahelp/org/zaproxy/zap/extension/pscanrules/resources/help/contents/pscanrules.html:524]: I'd suggest removing the spaces around the equals signs

File: addOns/pscanrules/src/main/javahelp/org/zaproxy/zap/extension/pscanrules/resources/help/contents/pscanrules.html [hunk 2]
```diff
@@ -409,34 +488,45 @@ <H2>VIEWSTATE</H2>
...
 This checks response headers for the presence of X-Debug-Token and X-Debug-Token-Link details.
 Which indicates the use/exposure of Symfony's Profiler. Symfony's Profiler provides access to a significant amount of information of interest to malicious individuals and Security Testers.
 <p>
 Latest code: <a href="https://github.com/zaproxy/zap-extensions/blob/main/addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/XDebugTokenScanRule.java">XDebugTokenScanRule.java</a>
+<br>
```
Comment (kingthorin) [inline addOns/pscanrules/src/main/javahelp/org/zaproxy/zap/extension/pscanrules/resources/help/contents/pscanrules.html:529]: Missed this last one

File: addOns/pscanrules/src/main/javahelp/org/zaproxy/zap/extension/pscanrules/resources/help/contents/pscanrules.html (hunk 1 above)
Comment (PariharAditya) [reply addOns/pscanrules/src/main/javahelp/org/zaproxy/zap/extension/pscanrules/resources/help/contents/pscanrules.html:524]: sure sir

File: addOns/pscanrules/src/main/javahelp/org/zaproxy/zap/extension/pscanrules/resources/help/contents/pscanrules.html [hunk 3]
```diff
@@ -409,34 +448,40 @@ <H2>VIEWSTATE</H2>
...
 This scan rule check for the Anti-MIME-Sniffing header X-Content-Type-Options and ensures it is set to 'nosniff'.<br>
 At MEDIUM and HIGH thresholds this rule does not alert on client or server error responses or redirects.<br> 
 At LOW threshold it will alert on all responses including errors and redirects.<br>
 <p>
 Latest code: <a href="https://github.com/zaproxy/zap-extensions/blob/main/addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/XContentTypeOptionsScanRule.java">XContentTypeOptionsScanRule.java</a>
-
-<H2>X-Debug-Token Information Leak</H2>
+<br>
+Alert ID: <a href="https://www.zaproxy.org/docs/alerts/10021/">10021</a>.
+<H2 id="id-10052">X-Debug-Token Information Leak</H2>
```
Comment (kingthorin) [inline addOns/pscanrules/src/main/javahelp/org/zaproxy/zap/extension/pscanrules/resources/help/contents/pscanrules.html:479]: X Chrome logger is already listed as 10052

I believe this is 10056

Comment (kingthorin) [reply addOns/pscanrules/src/main/javahelp/org/zaproxy/zap/extension/pscanrules/resources/help/contents/pscanrules.html:479]: It'd be good to have a blank before the heading

File: addOns/pscanrules/src/main/javahelp/org/zaproxy/zap/extension/pscanrules/resources/help/contents/pscanrules.html [hunk 4]
```diff
@@ -409,34 +448,40 @@ <H2>VIEWSTATE</H2>
...
 This checks response headers for the presence of X-Debug-Token and X-Debug-Token-Link details.
 Which indicates the use/exposure of Symfony's Profiler. Symfony's Profiler provides access to a significant amount of information of interest to malicious individuals and Security Testers.
 <p>
 Latest code: <a href="https://github.com/zaproxy/zap-extensions/blob/main/addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/XDebugTokenScanRule.java">XDebugTokenScanRule.java</a>
-
+<br>
+Alert ID: <a href="https://www.zaproxy.org/docs/alerts/10052/">10052</a>.
```
Comment (kingthorin) [inline addOns/pscanrules/src/main/javahelp/org/zaproxy/zap/extension/pscanrules/resources/help/contents/pscanrules.html:485]: X Chrome logger is already listed as 10052

I believe this is 10056

**Review by kingthorin — COMMENTED** [review]
I haven't checked the other entries/IDs

**Comment by PariharAditya** [comment]
any more changes ? todo ?

**Review by kingthorin — APPROVED** [review]
LGTM

File: addOns/pscanrules/src/main/javahelp/org/zaproxy/zap/extension/pscanrules/resources/help/contents/pscanrules.html [hunk 5]
```diff
@@ -248,42 +293,56 @@ <H2>Information Disclosure: Referrer</H2>
...
+Alert ID: <a href="https://www.zaproxy.org/docs/alerts/10040/">10040</a>.
 
-<H2>Modern Web Application</H2>
+<H2 id="id-10109">Modern Web Application</H2>
 This raises an informational alert if a site appears to be a modern web application.<br>
 It does not indicate any potential vulnerabilities but it could indicate that the ajax spider might be more effective at exploring this site compared to the traditional spider.
 <p>
 Latest code: <a href="https://github.com/zaproxy/zap-extensions/blob/main/addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/ModernAppDetectionScanRule.java">ModernAppDetectionScanRule.java</a>
+<br>
+Alert ID: <a href="https://www.zaproxy.org/docs/alerts/100109/">100109</a>.
```
Comment (thc202) [inline addOns/pscanrules/src/main/javahelp/org/zaproxy/zap/extension/pscanrules/resources/help/contents/pscanrules.html:336]: `10109`

File: addOns/pscanrules/src/main/javahelp/org/zaproxy/zap/extension/pscanrules/resources/help/contents/pscanrules.html [hunk 6]
```diff
@@ -304,8 +363,10 @@ <H2>PII Disclosure</H2>
 See: <a href="https://github.com/iannuttall/binlist-data">binlist-data</a> for more information.
 <p>
 Latest code: <a href="https://github.com/zaproxy/zap-extensions/blob/main/addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/PiiScanRule.java">PiiScanRule.java</a>
+<br>
+Alert ID: <a href="https://www.zaproxy.org/docs/alerts/10062/">10062</a>.
 
-<H2>Private Address Disclosure</H2>
+<H2 id="id-00002">Private Address Disclosure</H2>
```
Comment (thc202) [inline addOns/pscanrules/src/main/javahelp/org/zaproxy/zap/extension/pscanrules/resources/help/contents/pscanrules.html:369]: `id-2`

File: addOns/pscanrules/src/main/javahelp/org/zaproxy/zap/extension/pscanrules/resources/help/contents/pscanrules.html [hunk 7]
```diff
@@ -317,34 +378,44 @@ <H2>Private Address Disclosure</H2>
 repeated scans the "Context Alert Filters" add-on could be leveraged.
 <p>
 Latest code: <a href="https://github.com/zaproxy/zap-extensions/blob/main/addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/InfoPrivateAddressDisclosureScanRule.java">InfoPrivateAddressDisclosureScanRule.java</a>
+<br>
+Alert ID: <a href="https://www.zaproxy.org/docs/alerts/00002/">00002</a>.
```
Comment (thc202) [inline addOns/pscanrules/src/main/javahelp/org/zaproxy/zap/extension/pscanrules/resources/help/contents/pscanrules.html:382]: `2`

File: addOns/pscanrules/src/main/javahelp/org/zaproxy/zap/extension/pscanrules/resources/help/contents/pscanrules.html [hunk 8]
```diff
@@ -317,34 +378,44 @@ <H2>Private Address Disclosure</H2>
...
-<H2>Server Leaks Information via "X-Powered-By" HTTP Response Header Field(s)</H2>
+<H2 id="id-10037">Server Leaks Information via "X-Powered-By" HTTP Response Header Field(s)</H2>
 This checks response headers for the presence of X-Powered-By details.
 <p>
 Latest code: <a href="https://github.com/zaproxy/zap-extensions/blob/main/addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/XPoweredByHeaderInfoLeakScanRule.java">XPoweredByHeaderInfoLeakScanRule.java</a>
+<br>
+Alert ID: <a href="https://www.zaproxy.org/docs/alerts/10037/">10037</a>.
 
-<H2>Session ID in URL Rewrite</H2>
+<H2 id="id-00003">Session ID in URL Rewrite</H2>
```
Comment (thc202) [inline addOns/pscanrules/src/main/javahelp/org/zaproxy/zap/extension/pscanrules/resources/help/contents/pscanrules.html:408]: `id-3`

File: addOns/pscanrules/src/main/javahelp/org/zaproxy/zap/extension/pscanrules/resources/help/contents/pscanrules.html [hunk 9]
```diff
@@ -317,34 +378,44 @@ <H2>Private Address Disclosure</H2>
...
 This scan rule checks for the existence of session token type parameters being rewritten to the URL. 
 To help reduce false positives the rule checks the length of the token value,  if the value of the parameter 
 is not greater than 8 characters in length then the parameter is ignored 
 (i.e.: survey?sId=5 would not be flagged as vulnerable).
 <p>
 Latest code: <a href="https://github.com/zaproxy/zap-extensions/blob/main/addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/InfoSessionIdUrlScanRule.java">InfoSessionIdUrlScanRule.java</a>
+<br>
+Alert ID: <a href="https://www.zaproxy.org/docs/alerts/00003/">00003</a>.
```
Comment (thc202) [inline addOns/pscanrules/src/main/javahelp/org/zaproxy/zap/extension/pscanrules/resources/help/contents/pscanrules.html:416]: `3`

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Pull request merged**